### ISSUE TITLE
Does the reward_dim parameter actually do anything?

### ISSUE THREAD

**Issue by KZF-kzf** [TIGER-AI-Lab/EditReward#24]
I’ve noticed that whether I set it to dim1, dim2, or overall_detail, the output values remain exactly the same.
my code

```
import torch
from EditReward import EditRewardInferencer

CHECKPOINT_PATH = "EditReward-MiMo-VL-7B-SFT-2508"
CONFIG_PATH = "EditReward/config/EditReward-MiMo-VL-7B-SFT-2508.yaml"

_inferencer = None


def _get_inferencer(dim):
    global _inferencer
    if _inferencer is None:
        _inferencer = EditRewardInferencer(
            config_path=CONFIG_PATH,
            checkpoint_path=CHECKPOINT_PATH,
            device="cuda" if torch.cuda.is_available() else "cpu",
            reward_dim=dim,
        )
    return _inferencer


def compute_editreward_score(src_path: str, tgt_path: str, instruction: str, dim) -> float:
    inferencer = _get_inferencer(dim)
    with torch.no_grad():
        rewards = inferencer.reward(
            prompts=[instruction],
            image_src=[src_path],
            image_paths=[tgt_path]
        )
    print(rewards)
    print(dim)
    return rewards[0][0].item()


if __name__ == "__main__":
    img_src = "22_source.png"
    img_tgt = "22.png"
    prompt = "在古董车打开的行李箱里添加一只坐着的小棕狗。"
    score = compute_editreward_score(img_src, img_tgt, prompt, dim="dim1")
    print(f"EditReward result: {score:.4f}")
```

**Comment by KemingWu** [comment]
Thanks for reporting this.

What you are seeing is expected for the current released `EditReward-MiMo-VL-7B-SFT-2508` checkpoint.

A few clarifications:

- This checkpoint is configured with `rm_head_type="ranknet_multi_head"` and `pooling_strategy="mean"`.
- In that mode, the inference code always builds both `dim1` and `dim2` inputs internally, and then fuses the two heads into a single final output.
- So for the released multi-head checkpoint, `reward_dim` does not switch the returned score to an individual dimension-specific score.
- `overall_detail` is also not exposed as a separate third head in the current public inference path.

There is also one extra issue in your example: `_inferencer` is cached globally, so after the first initialization, later calls with a different `dim` will still reuse the same inferencer instance. However, even if you recreate the inferencer each time, the returned score will still be the same for this checkpoint because of the multi-head fusion described above.

One more note: the returned tensor has shape `[B, 2]`, but these two values are not `[dim1, dim2]`. For this checkpoint, they correspond to the fused reward statistics used by the uncertainty formulation, where `rewards[..., 0]` is the fused mean score and `rewards[..., 1]` is the log-uncertainty term.

So in short: with the current released MiMo checkpoint, getting the same value for `dim1`, `dim2`, and `overall_detail` is expected behavior.

If you need per-dimension outputs, we would need to expose the individual heads explicitly in the inference API rather than returning only the fused score.
